core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java

```diff
@@ -218,10 +224,11 @@ public static ColumnPage newPage(ColumnPageEncoderMeta columnPageEncoderMeta, in
         instance = newDoublePage(columnPageEncoderMeta, new double[pageSize]);
       } else if (DataTypes.isDecimal(dataType)) {
         instance = newDecimalPage(columnPageEncoderMeta, new byte[pageSize][]);
-      } else if (dataType == DataTypes.STRING
-          || dataType == BYTE_ARRAY
-          || dataType == DataTypes.VARCHAR
-          || dataType == DataTypes.BINARY) {
+      } else if (dataType == DataTypes.STRING ||
+          dataType == DataTypes.VARCHAR ||
+          dataType == DataTypes.BINARY) {
+        instance = new ByteBufferColumnPage(columnPageEncoderMeta, pageSize);
+      } else if (dataType == BYTE_ARRAY) {
```

Comment:
same as above

Reply:
It is not always for LV flow, for example, for local dictionary it uses UnsafeFixLengthColumnPage with store data type as BYTE_ARRAY